### PR TITLE
JDK-8320830: [AIX] Dont mix os::dll_load() with direct dlclose() calls

### PR DESCRIPTION
**Pull request by JoKern65**
We should not mix os::dll_load() with dlclose(), but should call os::dll_unload(). At the moment this is benign, but this prevents certain type of platforms specific workarounds inside os::dll_load() and is a prerequisite for these upcoming changes. 

There are two places I see where this happens, both are AIX specific:

```
os/aix/libodm_aix.cpp
51: dlclose(_libhandle);
58: if (_libhandle) { dlclose(_libhandle); }

os/aix/libperfstat_aix.cpp
117: dlclose(g_libhandle);
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320830](https://bugs.openjdk.org/browse/JDK-8320830): [AIX] Dont mix os::dll_load() with direct dlclose() calls (**Enhancement** - P4)

### Reviewers
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**) ⚠️ Review applies to [cb3df4fb](https://git.openjdk.org/jdk/pull/16846/files/cb3df4fb7131d0c21e1920a1da3dc70211fff50b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16846/head:pull/16846` \
`$ git checkout pull/16846`

Update a local copy of the PR: \
`$ git checkout pull/16846` \
`$ git pull https://git.openjdk.org/jdk.git pull/16846/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16846`

View PR using the GUI difftool: \
`$ git pr show -t 16846`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16846.diff">https://git.openjdk.org/jdk/pull/16846.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16846#issuecomment-1829660256)

### CODE REVIEW

**Review by tstuefe — APPROVED** [review]
Good + Trivial.

**Review by RealCLanger — APPROVED** [review]

File: src/hotspot/os/aix/libodm_aix.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@
 dynamicOdm::dynamicOdm() {
   const char* libodmname = "/usr/lib/libodm.a(shr_64.o)";
   char ebuf[512];
-  void* _libhandle = os::dll_load(libodmname, ebuf, sizeof(ebuf));
+  _libhandle = os::dll_load(libodmname, ebuf, sizeof(ebuf));
```
Comment (JoKern65) [inline src/hotspot/os/aix/libodm_aix.cpp:39]: _libhandle is a class member. Here it was accidentally defined with function local scope, so the class member _libhandle stayed uninitialized. As a result the dll_unload() in the destructor worked with a garbage value of the class member. This change fixes this problem.

Comment (tstuefe) [reply src/hotspot/os/aix/libodm_aix.cpp:39]: So the dllunload never worked, and we called dlclose with a garbage handle. Interesting that this never caused errors.

**Review by tstuefe — APPROVED** [review]
Still good.

**Pull request closed without merging**